src/ui-test/tests/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,92 @@
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+import {
+	EditorView,
+	VSBrowser,
+} from 'vscode-extension-tester';
+import { createIntegration, getIntegration, modifyCurrentFileToBeInvalid, prepareEmptyTestFolder, startIntegration } from '../utils/utils';
+import { cleanOutputView, sidebarIntegrationRemove } from '../utils/waitConditions';
+import * as uiTestConstants from '../utils/uiTestConstants';
+import { promiseFail, promiseSucceed } from '../utils/promiseUtils';
+
+const START_DEBUG_LABEL = uiTestConstants.startDebug;
+
+export function camelKJavaDebugTest() {
+
+	describe('Test Debug on Camel K Integrations from Side Bar', function () {
+
+		describe('Java Debug', function () {
+
+			const INTEGRATION_LABEL = 'java-debug-test';
+			const INTEGRATION_FILE = 'JavaDebugTest';
+
+			before(async function () {
+				this.timeout(uiTestConstants.TIMEOUT_60_SECONDS);
+				await createIntegration(INTEGRATION_FILE, 'Java', 'java');
+				await startIntegration(INTEGRATION_LABEL, 'Basic');
+			})
+
+			it('Check Java Debug available', async function () {
+				this.timeout(uiTestConstants.TIMEOUT_30_SECONDS);
+				const item = await getIntegration(INTEGRATION_LABEL);
+				const menu = await item.openContextMenu();
+				await promiseSucceed(VSBrowser.instance.driver.wait(() => menu.hasItem(START_DEBUG_LABEL), uiTestConstants.TIMEOUT_5_SECONDS));
+			});
+
+			after(async function () {
+				this.timeout(uiTestConstants.TIMEOUT_60_SECONDS);
+				await sidebarIntegrationRemove(VSBrowser.instance.driver, uiTestConstants.extensionName, INTEGRATION_LABEL);
+				await new EditorView().closeAllEditors();
+				await VSBrowser.instance.driver.wait(() => { return cleanOutputView(); });
+				await prepareEmptyTestFolder(uiTestConstants.testDir);
+			});
+
+		});
+
+		describe('No Java Debug on Invalid Files', function () {
+
+			const INTEGRATION_LABEL = 'java-debug-test-invalid';
+			const INTEGRATION_FILE = 'JavaDebugTestInvalid';
+
+			before(async function () {
+				this.timeout(uiTestConstants.TIMEOUT_60_SECONDS);
+				await createIntegration(INTEGRATION_FILE, 'Java', 'java');
+				await modifyCurrentFileToBeInvalid();
+				await startIntegration(INTEGRATION_LABEL, 'Basic');
+			});
+
+			it('Test Java Debugger Not Available On Invalid File', async function () {
+				this.timeout(uiTestConstants.TIMEOUT_30_SECONDS);
+				const item = await getIntegration(INTEGRATION_LABEL);
+				const menu = await item.openContextMenu();
+				await promiseFail(VSBrowser.instance.driver.wait(() => menu.hasItem(START_DEBUG_LABEL), uiTestConstants.TIMEOUT_5_SECONDS));
```

Comment:
Why not directly

```
VSBrowser.instance.driver.wait(() => !menu.hasItem(START_DEBUG_LABEL), uiTestConstants.TIMEOUT_5_SECONDS)
```

Reply:
First because that's not an assertion but counting on the code to work or fail whenever there is an exception.

Second because there are cases where promises could succeed or fail without having properly asserted the condition. We want to know if the wait command successfully completed and not just go forward.

There was more logic to it but I have to check the discussion because this was prompted by someone that requested it.

Reply:
Specially this one is impossible to assert if we use the wait command without using this promiseFail function.

Reply:
> First because that's not an assertion but counting on the code to work or fail whenever there is an exception.

the promiseFail is also not an assertion. it is based on the occurence of an Exception or not

> We want to know if the wait command successfully completed and not just go forward.

it is the case too with the

```
await VSBrowser.instance.driver.wait(() => !menu.hasItem(START_DEBUG_LABEL), uiTestConstants.TIMEOUT_5_SECONDS)
```

> Specially this one is impossible to assert if we use the wait command without using this promiseFail function.

i do not understand why. In Both cases are not using assertion for the failure.

If we want to have an assertion to provide more details on the reason of the failure than a timeout exception, it is possible to do something like:

```
try {
await VSBrowser.instance.driver.wait(() => !menu.hasItem(START_DEBUG_LABEL), uiTestConstants.TIMEOUT_5_SECONDS)
} catch (error) {
assert.fail('The start label should not be visible', e)
 }
```

Reply:
Which is actually a similar version to the definition of promiseFail function

const assertPromise = async (prom: any, succed = true) => {
    succed = succed ? true : false;
    try {
        prom = Array.isArray(prom) ? prom : [prom];
        await Promise.all(prom);
        assert.strictEqual(true, succed);
    }
    catch (ex: any) { assert.notStrictEqual(true, succed, ex); }
}

But more general and better named.

I can add another param to the assertPromise function to add what the message should be in case it doesn't work. Would that work it out?